python/paddle/nn/layer/container.py

```diff
@@ -0,0 +1,278 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+from ...fluid.dygraph.layers import Layer
+
+__all__ = ['LayerDict', ]
+
+
+class LayerDict(Layer):
+    """
+    LayerDict holds sublayers in the ordered dictionary, and sublayers it contains are properly registered.
+    Holded sublayers can be accessed like a regular ordered python dictionary. 
+
+    Parameters:
+        sublayers (iterable of Layer, optional): sublayers to hold
+
+    Examplex:
+        .. code-block:: python
+
+            import paddle
+            import numpy as np
+            from collections import OrderedDict
+
+            sublayers = OrderedDict([
+                ('conv1d', paddle.nn.Conv1D(3, 2, 3)),
+                ('conv2d', paddle.nn.Conv2D(3, 2, 3)),
+                ('conv3d', paddle.nn.Conv3D(4, 6, (3, 3, 3))),
+            ])
+
+            layers_dict = paddle.nn.LayerDict(sublayers=sublayers)
+
+            l = layers_dict['conv1d']
+
+            for k in layers_dict:
+                l = layers_dict[k]
+
+            len(layers_dict)
+            #3
+
+            del layers_dict['conv2d']
+            len(layers_dict)
+            #2
+
+            conv1d = layers_dict.pop('conv1d')
+            len(layers_dict)
+            #1
+
+            layers_dict.clear()
+            len(layers_dict)
+            #0
+
+    """
+
+    def __init__(self, sublayers=None):
+        super(LayerDict, self).__init__()
+        if sublayers is not None:
+            self._sub_layers.update(sublayers)
```

Comment:
here need call `self.update`, and the input should be checked in `update` method, otherwise here also need to check input

Reply:
Done